### DISCUSSION TITLE
How to debug 'GRAPHQL_VALIDATION_EXCEPTION'?

### DISCUSSION THREAD

**Discussion by berts83231** [directus/directus#10534] [Archive]
I have a simple query

```gql
query {
	available_coaches {
		id
		coach_id { email   }
	}}
```

It returns the error

```gql
{
  "errors": [
    {
      "message": "GraphQL validation error.",
      "extensions": {
        "code": "GRAPHQL_VALIDATION_EXCEPTION",
        "graphqlErrors": [
          {
            "message": "Field \"coach_id\" must not have a selection since type \"String\" has no subfields.",
            "locations": [
              {
                "line": 4,
                "column": 14
              }
            ]
          }
        ]
      }
    }
  ],
  "data": null
}
```

- It is an existing schema created outside of Directus.
- It is an issue on our staging DB but not our development DB
- Log level "debug" does not give enough info to find the issue.
- join available_coaches.coach_id -> users.id
- Directus 9.2.2 running in Docker (dev and staging)

What is the best way to debug this issue?

**Comment by rijkvanzanten** [comment]
In this particular case, the message already tells us what the problem is:

`Field \"coach_id\" must not have a selection since type \"String\" has no subfields.`

You're trying to nest a field under coach_id, but coach_id is a String, so that can't be done:

```graphql
# wrong
query {
	available_coaches {
		id
		coach_id { email   }
	}}

# right
query {
  available_coaches {
    id
    coach_id
  }
}
```

Now, the reason that coach ID is a String instead of a nested object is most likely because your current user doesn't have read access to the related collection.

**Reply by berts83231** [reply]
Thanks.

I deleted the Directus install / tables and tried again.

As you said, it might have been a permission issue I missed.

The following query works as expected now.

```gql
query {
	available_coaches {
		id
		sport_id { name }
                coach_id { full_name }
	}
}
```

**Reply by pikaju** [reply]
I also ran into this problem, thank God I found this thread, I was tearing my hair out.

**Reply by iksent** [reply]
I also had a problem with permissions
